intranet/apps/eighth/views/admin/scheduling.py

```diff
@@ -404,22 +406,44 @@ def transfer_students_action(request):
 
     num = source_act.members.count()
 
-    context = {"admin_page_title": "Transfer Students", "source_act": source_act, "dest_act": dest_act, "dest_unsignup": dest_unsignup, "num": num}
+    context = {
+        "admin_page_title": "Transfer Students",
+        "source_act": source_act,
+        "dest_act": dest_act,
+        "dest_unsignup": dest_unsignup,
+        "num": num,
+        "moved_students": None,
+    }
 
     if request.method == "POST":
         if dest_unsignup and not dest_act:
             source_act.eighthsignup_set.all().delete()
             invalidate_obj(source_act)
             messages.success(request, "Successfully removed signups for {} students.".format(num))
         else:
-            source_act.eighthsignup_set.update(scheduled_activity=dest_act)
+            duplicate_sign_up_users = []
+            duplicate_signups = []
+            with transaction.atomic():
+                for u in source_act.members.all():
+                    conflict_signup = u.eighthsignup_set.filter(scheduled_activity__block=dest_act.block)
+                    if conflict_signup.exists():
+                        duplicate_sign_up_users.append(u)
+                        duplicate_signups.append(conflict_signup[0])
+                lock_on(duplicate_sign_up_users)
```

Comment:
I would lock on **all** of the `User`s that are being transferred, to ensure that nothing happens concurrently.